xcrypto/aes.go

```diff
@@ -218,6 +218,38 @@ func (g *aesGCM) Seal(dst, nonce, plaintext, additionalData []byte) []byte {
 	return ret
 }
 
+func (g *aesGCM) SealWithRandomNonce(dst, nonce, plaintext, additionalData []byte) {
+	if len(nonce) != gcmStandardNonceSize {
+		panic("cipher: incorrect nonce length given to GCM")
+	}
+	if uint64(len(plaintext)) > ((1<<32)-2)*aesBlockSize || len(plaintext)+gcmTagSize < len(plaintext) {
+		panic("cipher: message too large for GCM")
+	}
+	if len(dst)+len(plaintext)+gcmTagSize < len(dst) {
+		panic("cipher: message too large for buffer")
+	}
+
+	if g.tls != cipherGCMTLSNone {
+		panic("cipher: TLS 1.2 and 1.3 modes do not support random nonce")
+	}
+
+	// Check delayed until now to make sure len(dst) is accurate.
+	if inexactOverlap(dst, plaintext) {
+		panic("cipher: invalid buffer overlap")
+	}
+	if anyOverlap(dst, additionalData) {
+		panic("crypto/cipher: invalid buffer overlap of output and additional data")
+	}
```

Comment:
This parameter validations looks almost good, but I would prefer to have exactly the same code and errors messages as in https://github.com/golang/go/blob/4f1146e6614676126d3810ea319e3f658d4f8116/src/crypto/internal/fips140/aes/gcm/gcm_nonces.go#L25-L39, just to make sure we are doing the same thing.

Reply:
fixed